### DISCUSSION TITLE
Convert multiple selected files to icon

### DISCUSSION THREAD

**Discussion by amymor** [ImageMagick/ImageMagick#1995] [General]
Hello
I just downloaded IM last week and i am new to it,

So first i learned about it and after a couple of day i created a batch and then conveted it to an EXE and then added it to PNG files context menu. so now I can convert a png to an icon by pressing 2 clicks. 
For example:
![2020-05-02 00_32_03](https://user-images.githubusercontent.com/54497554/81252882-fa68bf80-8fdb-11ea-906f-2d8e49f343c5.jpg)

But it can only convert one file. i wanna convert all selected file.

My first batch (1 line):
`D:\Downloads\Programs\#other\+Converter\ImageMagick-7.0.10-10-portable-Q16-x64\convert.exe "%~1" -define icon:auto-resize=256,64 "D:\Downloads\Programs\#desktop\#icons\x2ico\Icons\%~n1.ico"`

Created icons with this batch was so blurry. so i tried another method(5 line):
```
D:\Downloads\Programs\#other\+Converter\ImageMagick-7.0.10-10-portable-Q16-x64\convert.exe "%~1" -scale 256x256 "z1.ico"
D:\Downloads\Programs\#other\+Converter\ImageMagick-7.0.10-10-portable-Q16-x64\convert.exe "%~1" -scale 64x64 "z2.ico"
D:\Downloads\Programs\#other\+Converter\ImageMagick-7.0.10-10-portable-Q16-x64\convert.exe "z1.ico" "z2.ico" "D:\Downloads\Programs\#desktop\#icons\x2ico\Icons\%~n1.ico"
DEL "z1.ico"
DEL "z2.ico"
```
I also add both batchs with "%1" argument to right-click menu. also created an EXE from both and add it with "%1". but when I select multiple files and press Convert, always only one file is converted to icon

any idea?

Sry for my bad eng
Ty

**Comment by snibgo** [comment]
%1 is the first parameter to the BAT script. When you have multiple files, these are %2, %3 etc.

The usual method involves a shell loop with the Windows "shift" command.

**Reply by amymor** [reply]
Thanks
So if i use %9 and change all "%~1" in batch to "%~9", can i convert 9 selected pngs ?

**Reply by snibgo** [reply]
If you change %~1 to %~9, the script will convert just the ninth parameter.

**Reply by amymor** [reply]
So what should i do?
